gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -299,13 +339,40 @@ discreteElimination(const HybridGaussianFactorGraph &factors,
 #if GTSAM_HYBRID_TIMING
   gttic_(EliminateDiscrete);
 #endif
-  // NOTE: This does sum-product. For max-product, use EliminateForMPE.
-  auto result = EliminateDiscrete(dfg, frontalKeys);
+  // Check if separator is empty
```

Comment:
Seems very complicated. Does `frontalKeys().size() == dfg.keyVector().size()` not work?

Reply:
Is it possible that the number of keys are the same but the actual keys are different?
E.g. `{x0, x1, x2}` and `{x0, x1, y0}`?

I've updated to use the size check as suggested, but checking for explicit equality seemed wiser.

Reply:
No, not possible. Think about it for a bit :-)

Reply:
Yup I thought so, was curious.